crates/matrix-sdk-crypto/src/identities/user.rs

```diff
@@ -371,10 +402,87 @@ impl ReadOnlyUserIdentities {
 /// only contain a master key and a self signing key, meaning that only device
 /// signatures can be checked with this identity.
 #[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(from = "ReadOnlyUserIdentitySerializer", into = "ReadOnlyUserIdentitySerializer")]
```

Comment:
It might be cleaner to write custom `Serialize` and `Deserialize` implementations, but I'm not sure.

Reply:
I tried and for it to work properly I had to put a new field `version: Option<u8>` in the `ReadOnlyUserIdentity` model. 
I tried then with the helper pattern seen in other part of the code and it looks a bit better I think as it delegates serialization to a specific object and just needs standard `From` impl.